src/utils.ts

```diff
@@ -63,12 +77,15 @@ export function makeUrl(...args: string[]): string {
   return args.reduce((acc, cur) => urljoin(acc, cur));
 }
 
-const SKYLINK_MATCHER = "([a-zA-Z0-9_-]{46})";
-const SKYLINK_DIRECT_REGEX = new RegExp(`^${SKYLINK_MATCHER}$`);
-const SKYLINK_PATHNAME_REGEX = new RegExp(`^/?${SKYLINK_MATCHER}([/?].*)?$`);
-const SKYLINK_REGEXP_MATCH_POSITION = 1;
+export function parseSkylink(skylink: string, opts: any = {}): string {
+  let SKYLINK_MATCHER = "([a-zA-Z0-9_-]{46})";
+  if (opts.subdomain) {
+    SKYLINK_MATCHER = "([a-z0-9_-]{55}).";
+  }
```

Comment:
this function should be intelligent enough to determine that by itself, you shouldn't need to pass "opts.subdomain"

Reply:
We discussed that the function shouldn't try to be "smart", there could be e.g. non-skylink subdomains that happen to be 55 chars long.